### PR TITLE
MTP-2541 Apply grid to align svg, linktext and after element on discl…

### PR DESCRIPTION
**Pull request by petterkoding**
…osure link in menu

### CODE REVIEW

File: mt-kit/core/css/src/components/_disclosure.scss [hunk 1]
```diff
@@ -15,18 +15,38 @@
...
     width: 100%;
     margin: 0;
     padding: var(--disclosure-spacing) 0 var(--disclosure-spacing) var(--disclosure-spacing);
     overflow-wrap: anywhere;
+    &:not([href]) {
+      @include apply-icon($icon: $icon-carret-down-primary, $position: before) {
+        flex-shrink: 0;
+      }
+    }
+    &[href] {
```
Comment (lha047) [inline mt-kit/core/css/src/components/_disclosure.scss:30]: Is this specifically for the menu? If it is it should be with the rest of the menu stuff. Maybe there should be a separate menu style file

File: mt-kit/core/svelte/src/lib/svelte/content/MenuItems.svelte [hunk 2]
```diff
@@ -60,7 +60,7 @@
         <a
           href={item.url}
           rel={mapRelExternal(item.url)}
-          class="disclosure-header layout-flex layout-flex--center-vertical layout-full-width menu-level-1"
+          class="disclosure-header layout-flex layout-flex--center-vertical layout-full-width menu-level-1 align-items-center"
```
Comment (lha047) [inline mt-kit/core/svelte/src/lib/svelte/content/MenuItems.svelte:63]: If you need a grid fot this to work the way you want. Then use the grid layout classes. Consider if you should remove the disclosure-header class here. As it is not really a disclosure header

File: mt-kit/core/css/src/components/_disclosure.scss [hunk 3]
```diff
@@ -15,18 +15,38 @@
...
-    }
     display: flex;
-    align-items: center;
+    align-items: flex-start;
     gap: var(--spacer-xxx-small);
     width: 100%;
     margin: 0;
     padding: var(--disclosure-spacing) 0 var(--disclosure-spacing) var(--disclosure-spacing);
     overflow-wrap: anywhere;
+    &:not([href]) {
```
Comment (lha047) [inline mt-kit/core/css/src/components/_disclosure.scss:25]: There should be no need to exclude stuff with a href.

File: mt-kit/core/css/src/components/_disclosure.scss [hunk 4]
```diff
@@ -15,18 +15,38 @@
   }
 
   .disclosure-header {
```
Comment (lha047) [inline mt-kit/core/css/src/components/_disclosure.scss:17]: This should work for all the disclosures in general. If you need something special for the menu, place it in the menu styling.

**Review by lha047 — CHANGES_REQUESTED** [review]
See comments

File: mt-kit/core/css/src/components/_disclosure.scss [hunk 5]
```diff
@@ -15,18 +15,24 @@
...
     width: 100%;
     margin: 0;
     padding: var(--disclosure-spacing) 0 var(--disclosure-spacing) var(--disclosure-spacing);
     overflow-wrap: anywhere;
+
+    .title {
+      line-height: $text-with-icon-line-height;
+    }
+
+    &.align-items-center {
```
Comment (lha047) [inline mt-kit/core/css/src/components/_disclosure.scss:33]: Why not use grid on all the level 1 menu items?

File: mt-kit/core/css/src/components/_disclosure.scss [hunk 6]
```diff
@@ -15,18 +15,24 @@
...
-    align-items: center;
+    align-items: flex-start;
     gap: var(--spacer-xxx-small);
     width: 100%;
     margin: 0;
     padding: var(--disclosure-spacing) 0 var(--disclosure-spacing) var(--disclosure-spacing);
     overflow-wrap: anywhere;
+
+    .title {
+      line-height: $text-with-icon-line-height;
```
Comment (lha047) [inline mt-kit/core/css/src/components/_disclosure.scss:30]: Screenshot 2024-09-25 at 12 00 21
Look at the html implementation. This does not align all the stuff in the header. If you have chapter it looks like this.

**Review by lha047 — CHANGES_REQUESTED** [review]
Good :D See comments

File: mt-kit/core/css/src/components/_dropdown.scss [hunk 7]
```diff
@@ -321,6 +321,29 @@
       font-weight: var(--font-weight-bold);
     }
   }
+  .disclosure {
+    > a {
```
[the remark this answers is not in the document]
Comment (petterkoding) [reply mt-kit/core/css/src/components/_dropdown.scss:325]: moving it to .menu-level-1

File: mt-kit/core/css/src/components/_disclosure.scss (hunk 6 above)
Comment (petterkoding) [reply mt-kit/core/css/src/components/_disclosure.scss:30]: Designer has a plan to remove the icon, so it wont be a problem anymore. I will remove it

**Review by lha047 — APPROVED** [review]
good job 👍

**Pull request merged**